From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Nicolas Iooss <nicolas.iooss_linux@m4x.org>
Date: Fri, 15 Jan 2016 13:49:34 +0100
Subject: [PATCH] {BUG} drivers/staging/wilc1000/wilc_spi

clang:
/home/nicolas/projects/linux-patches/linux/drivers/staging/wilc1000/wilc_spi.c:123:34: error: tentative definition of variable with internal linkage has incomplete non-array type 'const struct wilc1000_ops'
      [-Werror,-Wtentative-definition-incomplete-type]
static const struct wilc1000_ops wilc1000_spi_ops;
                                 ^
/home/nicolas/projects/linux-patches/linux/drivers/staging/wilc1000/wilc_spi.c:123:21: note: forward declaration of 'struct wilc1000_ops'
static const struct wilc1000_ops wilc1000_spi_ops;
                    ^
/home/nicolas/projects/linux-patches/linux/drivers/staging/wilc1000/wilc_spi.c:123:34: error: tentative definition has type 'const struct wilc1000_ops' that is never completed
static const struct wilc1000_ops wilc1000_spi_ops;
                                 ^
/home/nicolas/projects/linux-patches/linux/drivers/staging/wilc1000/wilc_spi.c:123:21: note: forward declaration of 'struct wilc1000_ops'
static const struct wilc1000_ops wilc1000_spi_ops;
                    ^
---
 drivers/staging/wilc1000/wilc_spi.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/staging/wilc1000/wilc_spi.c b/drivers/staging/wilc1000/wilc_spi.c
index 86de50c9f7f5..90f36106c8d1 100644
--- a/drivers/staging/wilc1000/wilc_spi.c
+++ b/drivers/staging/wilc1000/wilc_spi.c
@@ -120,7 +120,7 @@ static u8 crc7(u8 crc, const u8 *buffer, u32 len)
 
 #define USE_SPI_DMA     0
 
-static const struct wilc1000_ops wilc1000_spi_ops;
+/*static const struct wilc1000_ops wilc1000_spi_ops; // Unused & unknown type? */
 
 static int wilc_bus_probe(struct spi_device *spi)
 {
-- 
